app/utils.js

```diff
@@ -16,15 +16,23 @@ aq.addFunction("utcsixhours", (x) => {
 // downsample the buoy points to approximately the desired number of points
 // TODO: have this handle multiple buoys in one pass
 const downsample = (data, numPoints, variables) => {
+  if (data.length === 0) {
+    return data;
+  }
+
   let full_dt = aq
     .from(data)
     .fold(variables, { as: ["variable", "value"] })
-    .filter("d.value");
+    .filter("d.value")
+    .orderby("station_name", "variable", "time")
+    .reify();
 
   if (full_dt.numRows <= numPoints) {
     return full_dt.objects();
   }
 
+  // TODO: what to do with non-numeric values??
```

Comment:
Throw an error? Do you expect non-numeric values when downsampling?

Reply:
There is a series of variables that are non-numeric, but we also can't do anything with them on the front end.  Right now arquero just says the `mean` is `null` if it gets downsampled.  Maybe I should just filter those variables out of the query string at the beginning...

Reply:
decided to filter them out - the download on the web app goes straight to erddap so there's not a great reason to ship the data from the API